src/packet.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <linux/inet.h>
+#include <linux/ip.h>
+#include <linux/tcp.h>
```

Comment:
Are these headers required here?
If not remove to implementation file (.c).

You can use forward declaration for tcp_hdr and ip_hdr:
```
struct tcphdr;
struct iphdr;
```